internal/services/loganalytics/log_analytics_linked_storage_account_resource.go

```diff
@@ -134,7 +134,7 @@ func resourceLogAnalyticsLinkedStorageAccountRead(d *pluginsdk.ResourceData, met
 	d.Set("resource_group_name", id.ResourceGroup)
 	d.Set("workspace_resource_id", parse.NewLogAnalyticsWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName).ID())
 	if props := resp.LinkedStorageAccountsProperties; props != nil {
-		d.Set("data_source_type", string(props.DataSourceType))
+		d.Set("data_source_type", strings.ToLower(string(props.DataSourceType)))
```

Comment:
The validation function is wrong, we shouldn't be making every official swagger value all lowercase. to correctly fix this can we update the validation function to use the correct casing? i presume ingestion will also need to be fixed?

Reply:
+1 we can normalize the casing in the Read function here, making the field case-insensitive temporarily - then (in the future) fix this field so that it's case-sensitive in 4.0 (using a feature-flag)

Reply:
The issue is releated to service API issue https://github.com/Azure/azure-rest-api-specs/issues/20619, case sensitivity differs with locations, and changing request case doesn't affect the response. I think current fix is to ingore Enum case sensivity. And after the service fix their API issue, we should add a fix in 4.0 with feature-flag.

Reply:
Thats ok, we should still normalize the response constants to what we expect and then flag case insensivity to the 4.0 flag